File: app/src/main/java/com/doapps/habits/helper/PicassoRoundedTransformation.java
package com.doapps.habits.helper;

import android.graphics.Bitmap;
import android.graphics.BitmapShader;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.RectF;
import android.graphics.Shader;

import com.squareup.picasso.Transformation;

public class PicassoRoundedTransformation implements Transformation {
    private int radius;

    @Override
    public Bitmap transform(final Bitmap source) {
        final Paint paint = new Paint();
        radius = source.getHeight() >> 1;
        paint.setAntiAlias(true);
        paint.setShader(new BitmapShader(source, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP));

        Bitmap output = Bitmap.createBitmap(source.getWidth(), source.getHeight(), Bitmap.Config.ARGB_8888);
        Canvas canvas = new Canvas(output);
        canvas.drawRoundRect(new RectF(0, 0, source.getWidth(),
                source.getHeight()), radius, radius, paint);

        source.recycle();

        return output;
    }

    @Override
    public String key() {
        return "rounded(radius=" + radius + ")";
    }
}

File: app/src/main/java/com/doapps/habits/helper/PicassoRoundedTransformation.kt
package com.doapps.habits.helper

import android.graphics.*
import com.squareup.picasso.Transformation

class PicassoRoundedTransformation : Transformation {
  private var radius: Int = 0

  override fun transform(source: Bitmap): Bitmap {
    val paint = Paint()
    radius = source.height shr 1 // shift right
    paint.isAntiAlias = true
    paint.shader = BitmapShader(source, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP)

    val output = Bitmap.createBitmap(source.width, source.height, Bitmap.Config.ARGB_8888)
    val canvas = Canvas(output)
    canvas.drawRoundRect(RectF(0f, 0f, source.width.toFloat(),
        source.height.toFloat()), radius.toFloat(), radius.toFloat(), paint)

    source.recycle()

    return output
  }

  override fun key() = "rounded(radius=$radius)"
}
